### DISCUSSION TITLE
Project setup on Tina Cloud

### DISCUSSION THREAD

**Discussion by georgibakken** [tinacms/tinacms#3240] [Help] [answered]
Hello,

I am trying to setup a project on Tina Cloud, and have gotten 2/4 of the checklist. I am using the [docs-starter](https://github.com/tinacms/tina-docs-starter). I have added the environment variables to my `.env file`, taken from Tina Cloud.

I have not deployed the tina docs starter anywhere, just running on localhost and configured the url to be `http://localhost:3000`.

My problem is to configure the last two points in the checklist:
<img width="528" alt="image" src="https://user-images.githubusercontent.com/16558100/192625443-917270fc-032b-4229-aabb-5355a88bc6a0.png">

I believed setting the environment variables would be enough. Do I have to deploy the tina project to the cloud? If so, does it have to be netlify or Vercel, or could I deploy it to Azure? 

Thank you

**Comment by kldavis4** [comment]
You can run it locally to login and make a commit. It sounds like there might be a problem with the environment variables defined in `.env` - the important ones are: `NEXT_PUBLIC_TINA_CLIENT_ID`, `TINA_TOKEN` and `NEXT_PUBLIC_TINA_BRANCH` - can you verify that those are defined? If you have those correctly set in the `.env` and start the server locally (`yarn dev`), you should be prompted to login when you hit http://localhost:3000/admin

**Comment by georgibakken** [comment]
Thanks @kldavis4 for the quick reply!

Yes, I can verify that those are three defined. 

This is the following screen, when I try http://localhost:3000/admin/

<img width="701" alt="image" src="https://user-images.githubusercontent.com/16558100/192627945-6a7628d5-0cb3-4daf-86cf-179ba1351b07.png">

I am guessing I should have been prompted to be logged in here instead?

**Comment by kldavis4** [comment]
That's the correct screen - just click Edit With Tina and you should get the login screen

**Comment by georgibakken** [comment]
Unfortunately the next screen is just straight to the edit mode, in Local Mode it says.

I can also see on Tina Cloud that my token has never been used, so most likely yes, there's something off with the variables.

**Comment by kldavis4** [comment, marked as answer]
It looks like the current behavior is that `yarn dev` will always run local mode. If you run `yarn build && yarn start` it should connect to tina cloud.

**Reply by georgibakken** [reply]
Yes! Thank you! that was it, it works

**Reply by leafarafael** [reply]
Can I use tinacms build && hugo in Github Actions ?

**Reply by kldavis4** [reply]
yes, https://tina.io/docs/introduction/tina-init/#hugo-example

**Reply by leafarafael** [reply]
@kldavis4 Hi
I actually posted my issue here:
https://github.com/tinacms/tinacms/discussions/3586
It was in the docs that it should be tinacms build && hugo but I think I'm missing something.
